packages/cosmic-swingset/src/chain-main.js

```diff
@@ -72,7 +74,26 @@ const toNumber = specimen => {
  * @typedef {object} CosmosSwingsetConfig
  * @property {string} [slogfile]
  * @property {number} [maxVatsOnline]
+ * @property {'debug' | 'operational'} [vatSnapshotRetention]
+ * @property {'archival' | 'operational'} [vatTranscriptRetention]
  */
+const SwingsetConfigShape = M.splitRecord(
+  {},
+  {
+    slogfile: M.string(),
+    maxVatsOnline: M.number(),
+    vatSnapshotRetention: M.or('debug', 'operational'),
+    vatTranscriptRetention: M.or('archival', 'operational'),
+  },
+  M.any(),
```

Comment:
This `M.any()` allows `resolvedConfig` to include unknown fields, which supports changes on the Go side but can also let mistakes silently slip through without affecting the operation of cosmic-swingset. I could be persuaded towards `{}` strictness here.

Reply:
given that we only deploy Go-side changes at the same (chain-halting upgrade) time as the JS-side, it's probably better to be strict, and not allow any unknown properties

Reply:
Actually, is there any reason that a known property might be missing? Should we move it into the first slot, making them mandatory? I didn't follow the Go code well enough to see where the inbound structure is created, and whether the properties would be omitted if e.g. the `app.toml` didn't include a `[swingset]` section.

Maybe sim-chain comes through here without a complete record?